### PR TITLE
[MRG] Turn off datatype check 

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Closes: #743 

Should I add a test even?

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

File: mne_bids/path.py [hunk 1]
```diff
@@ -557,7 +557,8 @@ def update(self, *, check=None, **kwargs):
                 continue
 
             if key == 'datatype':
-                if val is not None and val not in ALLOWED_DATATYPES:
+                if val is not None and val not in ALLOWED_DATATYPES \
```
Comment (hoechenberger) [inline mne_bids/path.py:560]: Can we not drop the `None` check?

Comment (hoechenberger) [reply mne_bids/path.py:555]: oh, never mind.

File: doc/whats_new.rst [hunk 2]
```diff
@@ -47,6 +47,7 @@ Enhancements
 - Arbitrary file names can now be represented as a `BIDSPath`` by passing the entire name as ``suffix`` and setting ``check=False``, by `Adam Li`_ (:gh:`729`)
 - Add support for MNE's flux excitation channel (``exci``), by `Maximilien Chaumon`_ (:gh:`728`)
 - :meth:`mne_bids.BIDSPath.match` gained a new parameter ``check``; when setting ``check=True``, ``match()`` will only return paths that conform to BIDS, by `Richard Höchenberger`_ (:gh:`726`)
+- MNE datatypes that are currently unsupported in MNE-BIDS (e.g. ``dwi``, ``func``) can now be used in :class:`mne_bids.BIDSPath` by setting ``check=False``, by `Adam Li`_ (:gh:`744`)
```
Comment (hoechenberger) [inline doc/whats_new.rst:50]:
```suggestion
- Datatypes that are currently unsupported by MNE-BIDS (e.g. ``dwi``, ``func``) can now be used in :class:`mne_bids.BIDSPath` by setting ``check=False``, by `Adam Li`_ (:gh:`744`)
```

**Comment by adam2392** [comment]
I can add more documentation to the description of check actually. 

I feel like BIDSpath is evolving into a very generic usage class not just in MNE (which is awesome imo). 

But at the same time we want to allow flexible usage and error checking if they venture away from MEG/EEG/IEEG.

**Comment by hoechenberger** [comment]
> I can add more documentation to the description of check actually.

Sure, go for it!

But apparently there's also still some tests that will need our attention…

**Comment by hoechenberger** [comment]
@adam2392 Do you insist on having this in 0.7 or can we push this to 0.8?

**Comment by adam2392** [comment]
Can I push after I return from the gym to squeeze into 0.7? :p

**Comment by hoechenberger** [comment]
Alright, enjoy your exercise!

💪 🏋️

**Comment by adam2392** [comment]
This should be good to go once CI works @hoechenberger

**Comment by adam2392** [comment]
rip: looks ugly https://4339-89170358-gh.circle-artifacts.com/0/dev/generated/mne_bids.BIDSPath.html#mne_bids.BIDSPath

will fix.

**Comment by hoechenberger** [comment]
Thanks @adam2392!

**Pull request merged**